web/src/pages/TokenVerification/TokenVerification.tsx

```diff
@@ -0,0 +1,24 @@
+import React, { useEffect } from 'react';
+import { useNavigate, Outlet } from 'react-router-dom';
+import Cookies from 'js-cookie';
+
+function PrivateRoute() {
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const token = Cookies.get('Token');
+    const expirationDate = Cookies.get('ExpirationDate');
+
+    if (!token) {
+
+      navigate('/login');
+    } else {
+      // Si un token est présent et la date d'expiration n'est pas dépassée, autoriser l'accès à la route
```

Comment:
add a pop-up that says 'expired token' and redirect to main page here

Reply:
actually, the more i read the code, the more i don't understand

You're getting an `ExpirationDate` value from the cookies that is never set, and then never use it? 
It would be simpler to wrap around the API requests and redirect to `/login` if the call fails.

You may find a simple example here:
```js
export const genericRequest = async (
  httpVerb,
  route,
  body = {},
) => {
  const authErrorMessages = [
    'Invalid token.',
    'User is not active',
    'The token is expired.',
    'Authentication credentials were not provided.',
  ];

  try {
    const response = await axios({
      method: httpVerb,
      url: `${API_URL}${route}`,
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Token ${Cookies.get('Token')}`, 
      },
      data: body,
    });
    return response.data;
  } catch (err) {
    if (err.response !== undefined && err.response.data !== undefined) {
      if (authErrorMessages.includes(err.response.data.detail)) {
        // maybe you can also add a pop-up here that displays the error message
        navigate('/login');
      }
    }
    return error: err.response.data;
  }
};
```

Then, for every API call in the front-end, you can just do
```js
const data = await genericRequest(
    'get',
    `/an-API-route-from-back-end?token=${token}`,
  );
```